src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,45 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.BulldozerProducer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.ExcavatorProducer;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends T> type) {
+        MachineProducer<? extends T> producer;
+
+        if (type == Bulldozer.class) {
+            producer = (MachineProducer<? extends T>) new BulldozerProducer();
+        } else if (type == Excavator.class) {
+            producer = (MachineProducer<? extends T>) new ExcavatorProducer();
+        } else if (type == Truck.class) {
+            producer = (MachineProducer<? extends T>) new TruckProducer();
+        } else {
+            return new ArrayList<>();
+        }
+
+        return (List<Machine>) producer.get();
+    }
```

Comment:
Do not use over-complicated `if-else` constuction, bette to use `switch` here, for example:

```java
    @Override
    public List<Machine> getAll(Class<? extends Machine> type) {
        List<? extends Machine> machines;
        switch (type.getSimpleName()) {
            case "Bulldozer" -> machines = new BulldozerProducerImpl().get();
            case "Excavator" -> machines = new ExcavatorProducerImpl().get();
            case "Truck" -> machines = new TruckProducerImpl().get();
            default -> {
                return Collections.emptyList();
            }
        }
        return new ArrayList<>(machines);
    }
```